### PR TITLE
Cache `data.` calls to greatly speed everything up.

### PR DESCRIPTION
**Pull request by kylorhall**
For rspec tests primarily.

Thoughts?  Might be doing a few of the caches oddly 🤷‍♂️

### CODE REVIEW

File: helpers/middleman/blog.rb [hunk 1]
```diff
@@ -4,25 +4,30 @@
...
-      data.blog.posts
-    end
   end
 
   def post_url(post)
     return unlocalized_url("blog/#{BlogHelper::url_slug(post)}")
   end
 
   def blog_categories
+    return @blog_categories if @blog_categories
```
Comment (alto) [inline helpers/middleman/blog.rb]: This is so...

The Ruby way would be to

```ruby
def blog_categories
  @blog_categories ||= begin
    ..
  end
end
```

File: helpers/middleman/partners.rb [hunk 2]
```diff
@@ -74,6 +80,6 @@ def partners_categories(lang = default_locale_obj[:lang], withIndex: false)
     end
 
     # Stick the all category to the top.
-    return array.sort{ |a, b| sort_categories(a, b) }
+    @partners_categories[lang][withIndex] ||= array.sort{ |a, b| sort_categories(a, b) }
```
Comment (alto) [inline helpers/middleman/partners.rb]: Don't you want to `return`?

File: helpers/middleman/partners.rb [hunk 3]
```diff
@@ -38,6 +40,10 @@ def categories_collection(lang = default_locale_obj[:lang])
 
   # localize_value should always have a third argument: fallback_locale!
   def partners_categories(lang = default_locale_obj[:lang], withIndex: false)
+    @partners_categories ||= {}
+    @partners_categories[lang] ||= {}
+    return @partners_categories[lang][withIndex] if @partners_categories[lang][withIndex]
```
Comment (alto) [inline helpers/middleman/partners.rb]: See above (The Ruby Way ™).

File: helpers/middleman/partners.rb (hunk 2 above)
Comment (kylorhall) [reply helpers/middleman/partners.rb]: I'll add it back in, thanks! 👍

File: helpers/middleman/blog.rb (hunk 1 above)
Comment (kylorhall) [reply helpers/middleman/blog.rb]: Ah, alright.  the `begin; end` was what I missed – didn't want to run the other crap.

File: helpers/middleman/partners.rb (hunk 3 above)
Comment (kylorhall) [reply helpers/middleman/partners.rb]: How can I do `@partners_categories[lang][withIndex] ||= ` if `@partners_categories` is `nil`?

Also, is `@partners_categories[nil]` or `@partners_categories[false]` a valid key index?

Comment (alto) [inline helpers/middleman/partners.rb]: This should be `@partners_categories[lang][!!withIndex]` actually, as for hashes `hash[nil]` is not the same as `hash[false]`.

Comment (alto) [reply helpers/middleman/partners.rb]: You cannot. You always have to `@categories_collection ||= {}` as a first step.

**Pull request merged**